engine/orchestration/src/com/cloud/vm/VirtualMachineManagerImpl.java

```diff
@@ -1472,6 +1473,12 @@ private void advanceStop(final VMInstanceVO vm, final boolean cleanUpEvenIfUnabl
                 _workDao.update(work.getId(), work);
             }
             return;
+        } else {
+            HostVO host = _hostDao.findById(hostId);
+            if (!cleanUpEvenIfUnableToStop && vm.getState() == State.Running && host.getResourceState() == ResourceState.PrepareForMaintenance) {
+                s_logger.debug("Host in PrepareForMaintenance state - Failed to stop VM with id: " + vm.getId());
```

Comment:
hello @sureshanaparti.

How about you extract the content in the 'if' at line 1478 to an "isHostPreparingForMaintenance" method or something like that? it will explain better the if content, and you can do a test case to this method too

Ty.